exercise/permissions.py

```diff
@@ -31,6 +31,10 @@ def is_object_visible(self, request, view, exercise):
             self.error_msg(_("The exercise is not currently visible."))
             return False
 
+        if exercise.is_submittable and not exercise.course_module.have_exercises_been_opened():
+            self.error_msg(_("The exercise opens for submissions in {}.".format(exercise.course_module.opening_time)))
```

Comment:
I suspect that the date is formatted in the wrong way in the translated string. Shouldn't the `format` method be called on the string only after the `_` function has been called? Now you are trying to translate a string that contains a specific date value. Does it really work at all when you test it? The trivial fix is to move the parenthesis:
`_("The exercise opens for submissions in {}.").format(exercise.course_module.opening_time)`

Reply:
Moved the format out of the function.